src/map/map.ts

```diff
@@ -128,38 +128,89 @@ function arrayMaxMin(values: number[]): {max: number, min: number} {
     return {max, min};
 }
 
-/** HTML element holding settings for a given axis (x, y, z, color) */
-class AxisSetting {
+interface AxisPresets {    
     /// Which property should we use for this axis
-    public property: HTMLSetting<'string'>;
+    property?: string;
     /// Which scale (linear/log) should we use
-    public scale: HTMLSetting<'string'>;
+    scale?: string;
     /// The minimal value for this axis
-    public min: HTMLSetting<'number'>;
+    min?: number;
     /// The maximal value for this axis
-    public max: HTMLSetting<'number'>;
+    max?: number;
+}
+
+const AxisDefaults = {
```

Comment:
Being a global variable, I would use `SCREAMING_SNAKE_CASE` instead of `CapitalCamelCase` which is usually reserved for types